Mathlib/Analysis/Polynomial/CauchyBound.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.Order.Nonneg.GeomSum
+import Mathlib.Algebra.Polynomial.Eval
+import Mathlib.Algebra.Polynomial.Monic
+import Mathlib.Analysis.Normed.Field.Basic
+
+/-!
+# Cauchy's bound on polynomial roots.
```

Comment:
Please write a paragraph or two that mention the main result and the formula for the Cauchy's bound.